target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl931x.c

```diff
@@ -1566,50 +1567,87 @@ static void rtldsa_931x_led_init(struct rtl838x_switch_priv *priv)
 		return;
 	}
 
+	for (int set = 0; set < 4; set++) {
+		char set_name[16] = {0};
+		u32 set_config[4];
+		int leds_in_this_set = 0;
+
+		/*Reset LED set configuration */
+		sw_w32(0, RTL931X_LED_SETX_0_CTRL(set));
+		sw_w32(0, RTL931X_LED_SETX_1_CTRL(set));
+
+		/*
+		 * Each LED set has 4 number of leds, and each LED is configured with 16 bits
+		 * So eacch 32 bit register holds configuration for 2 leds
+		 * and therefore each set requires 2 registers for configuring 4 LEDs
+		 */
+		snprintf(set_name, sizeof(set_name), "led_set%d", set);
+		leds_in_this_set = of_property_count_u32_elems(node, set_name);
+
+		if (leds_in_this_set == 0 || leds_in_this_set > sizeof(set_config)) {
+			pr_err("%s led_set configuration invalid skipping over this set\n", __func__);
+			continue;
+		}
+
+		if (of_property_read_u32_array(node, set_name, set_config, leds_in_this_set)) {
+			break;
+		}
+
+		/* Write confgiruation as per number of LEDs */
+		for (int i = 0, led = leds_in_this_set - 1; led >= 0; led--, i++) {
+			sw_w32_mask(0xffff << RTL931X_LED_SET_LEDX_SHIFT(led),
+						(0xffff & set_config[i]) << RTL931X_LED_SET_LEDX_SHIFT(led),
+						RTL931X_LED_SETX_LEDY(set, led));
+		}
+	}
+
+	pm_auto_detect = of_property_read_u64(node, "forced_port_mask", &pm_forced);
```

Comment:
You use it like a bool, so maybe just make it a bool. And then change `pm_auto_detect == 0` to `!pm_auto_detect` and `pm_auto_detect != 0` to `pm_auto_detect`